packages/components/src/autocomplete/types.ts

```diff
@@ -187,7 +187,7 @@ export type UseAutocompleteProps = {
 	 * A function to be called when an option is selected to insert into the
 	 * existing text.
 	 */
-	onChange: ( value: string ) => void;
+	onChange: ( value: RichTextValue ) => void;
```

Comment:
At the very least, `RichTextValue` is what the actual callback receives in the rich-text package.